pkgs/tools/filesystems/dbxfs/default.nix

```diff
@@ -0,0 +1,101 @@
+{ lib, fetchgit, python3Packages, fusePackages }:
+
+let
+  block-tracing = python3Packages.buildPythonPackage rec {
+    pname = "block_tracing";
+    version = "1.0.1";
+
+    src = python3Packages.fetchPypi {
+      inherit version;
+      inherit pname;
+      sha256 = "9faa009a702a8f2f605ebb78314d5ca2a2a93543d061038a3d3a978c93385e68";
+    };
+  };
+  privy = python3Packages.buildPythonPackage rec {
+    pname = "privy";
+    version = "6.0.0";
+    format = "wheel";
+
+    src = python3Packages.fetchPypi {
+      inherit version;
+      inherit pname;
+      format = "wheel";
+      sha256 = "e68679bb4006ce83206d9a7af1158cf4d56a2ed6861ee39276907be618dfb8d9";
+    };
+
+    propagatedBuildInputs = [
+      python3Packages.cryptography
+      python3Packages.argon2_cffi
+    ];
+  };
+  fusepyng = python3Packages.buildPythonPackage rec {
+    pname = "fusepyng";
+    version = "1.0.7";
+
+    src = python3Packages.fetchPypi {
+      inherit version;
+      inherit pname;
+      sha256 = "sha256-i09u+B6GAPI9p1CRaayyYVWC7xFtRqKhrUt+Uw2PiZ8=";
+    };
+
+  };
+  userspacefs = python3Packages.buildPythonPackage rec {
+    pname = "userspacefs";
+    version = "2.0.5";
+
+    src = python3Packages.fetchPypi {
+      inherit version;
+      inherit pname;
+      sha256 = "sha256-XW9f+m05SI8cdDfF6q6Pq/SRbKzqtIOjFzoY5nibGGw=";
+    };
+
+    buildInputs = [
+      fusePackages.fuse_2
+    ];
+
+    # crashes looking for libfuse in build_ext (?)
+    doCheck = false;
+
+    propagatedBuildInputs = [
+      fusepyng
+    ];
+  };
+in
+python3Packages.buildPythonApplication rec {
+  pname = "dbxfs";
+  version = "1.0.58";
+
+  src = fetchgit {
+    url = "https://thelig.ht/code/dbxfs";
+    deepClone = true; # this repo doesn't support shallow cloning
```

Comment:
Do we have a source tarball we could fetch?

Reply:
I don't think so... this person runs their own git server, I don't see any download options.

Reply:
We _could_ probably use pypi instead of building from source though? https://pypi.org/project/dbxfs/
Not worth it IMHO...

Reply:
Please fetch the pypi tarball that is found here https://pypi.org/project/dbxfs/#files . This is totally worth it because cloning the git repo without shallow and tags takes a quite a few seconds and downloading the tarball should be almost instant.

Reply:
Done in https://github.com/NixOS/nixpkgs/pull/155134/commits/2ed350741f44677c2277f9c45ad4ae97f0dfd054